packages/swapper/src/swappers/PortalsSwapper/getPortalsTradeQuote/getPortalsTradeQuote.ts

```diff
@@ -120,10 +119,24 @@ export async function getPortalsTradeQuote(
       validate: true,
       swapperConfig,
     }).catch(async e => {
-      // If validation fails, fire two more quotes:
-      // 1. a quote with validation enabled, but using a well-funded address to get a rough gasLimit estimate
-      // 2. another quote with validation disabled, to get an actual quote
+      // If validation fails, fire 3 more quotes:
+      // 1. a quote estimate (does not require approval) to get the optimal slippage tolerance
+      // 2. a quote with validation enabled, but using a well-funded address to get a rough gasLimit estimate
+      // 3. another quote with validation disabled, to get an actual quote (using the user slippage, or the optimal from the estimate)
       console.info('failed to get Portals quote with validation enabled', e)
+
+      // Use the quote estimate endpoint to get the optimal slippage tolerance
+      const quoteEstimateResponse = await fetchPortalsTradeEstimate({
```

Comment:
Confirmed she's the goer!

<img width="482" alt="image" src="https://github.com/user-attachments/assets/102ebce5-0b2e-45d9-8429-aaed41dd1c50">